drivers/regulator/regulator_npm1300.c

```diff
@@ -512,16 +519,55 @@ int regulator_npm1300_common_init(const struct device *dev)
 	return 0;
 }
 
+static int get_enabled_reg(const struct device *dev, uint8_t base, uint8_t offset, uint8_t mask,
+			   bool *enabled)
+{
+	const struct regulator_npm1300_config *config = dev->config;
+	uint8_t data;
+
+	int ret = mfd_npm1300_reg_read(config->mfd, base, offset, &data);
+
+	if (ret == 0) {
+		*enabled = (data & mask) != 0U;
+	}
+
+	return ret;
```

Comment:
Could you swap the condition around and have the success case inline instead and have an explicit `return 0;` at the end? Makes it dead obvious to understand imo, that would be:

```c
int ret = mfd_npm1300_reg_read(config->mfd, base, offset, &data);
if (ret != 0) {
        return ret;
}

*enabled = (data & mask) != 0U; 

return 0;
```